test/view/study/study_list_screen_test.dart

```diff
@@ -0,0 +1,694 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/testing.dart';
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/view/study/study_list_screen.dart';
+
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+void main() {
+  group('Study list screen', () {
+    final mockClient = MockClient((request) {
+      if (request.url.path == '/study/all/hot') {
+        if (request.url.queryParameters['page'] == '1') {
+          return mockResponse(kStudyAllHotPage1Response, 200);
+        }
+        if (request.url.queryParameters['page'] == '2') {
+          return mockResponse(kStudyAllHotPage2Response, 200);
+        }
+        return mockResponse(
+          emptyPage(int.parse(request.url.queryParameters['page']!)),
+          200,
+        );
+      }
+      if (request.url.path == '/study/search') {
+        if (request.url.queryParameters['q'] == 'Magnus') {
+          return mockResponse(
+            '''
+{
+  "paginator": {
+    "currentPage": 1,
+    "maxPerPage": 16,
+    "currentPageResults": [
+      {
+        "id": "g26XbGpT",
+        "name": "Magnus Carlsen Games",
+        "liked": false,
+        "likes": 1,
+        "updatedAt": 1723817543350,
+        "owner": {
+          "name": "tom-anders",
+          "id": "tom-anders"
+        },
+        "chapters": [
+          "Chapter 1",
+          "Chapter 2"
+        ],
+        "topics": [ ],
+        "members": [ ]
+      }
+    ],
+    "previousPage": null,
+    "nextPage": null,
+    "nbResults": 1,
+    "nbPages": 1
+  }
+}
+              ''',
+            200,
+          );
+        }
+      }
+      return mockResponse('', 404);
+    });
+
+    testWidgets('Scrolling down loads next page', (WidgetTester tester) async {
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const StudyListScreen(),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) {
+            return LichessClient(mockClient, ref);
+          }),
+        ],
+      );
+      await tester.pumpWidget(app);
+      await tester.pumpAndSettle();
```

Comment:
Why is `pumpAndSettle` needed here? it is usually needed when animations need to finish, but I prefer put a comment each time we use `pump()` or `pumpOrSettle` to explain what are we waiting for, to help future readers.

Reply:
ah this one is needed, I added a comment

Reply:
(although `pump` seems to also be sufficient)